xbmc/filesystem/NFSDirectory.h

```diff
@@ -8,9 +8,13 @@
 
 #pragma once
 
+#include "FileItem.h"
+#include "FileItemList.h"
```

Comment:
FileItem.h actually needs not to be included in this header file.

For `CFileItemPtr` use the native type instead of the typedef from FileItem.h . Include <memory>, put `class CFileItem;` in this header and replace `CFileItemPtr`with `std::shared_ptr<CFileItem>`.